src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type T = {
+  [key: string]: any;
+};
+
+export function groupByKey(items: T[], key: string): GroupsMap<T> {
+  const groupedElements: GroupsMap<T> = {};
+
+  const keyValues: string[] = [...items].map((el) => el[key]);
```

Comment:
```suggestion
  const keyValues: string[] = items.map((el) => el[key]);
```
it's redundant 'cause map and filter do not mutate the array